MHV account creation and terms to master

aub opened the pull request.

aub committed:
getting started on accepting terms

aub committed:
fix lint issues

aub committed:
Merge branch 'master' into acct

aub committed:
put the terms acceptance thing back

aub committed:
display the accept terms prompt

aub committed:
hardcode in the content for now

bshyong committed:
fix DOM nesting

bshyong committed:
add actions reducers for MHV terms acceptance checking

bshyong committed:
show child component if terms are accepted

aub committed:
basically working terms acceptance

aub committed:
fix

aub committed:
implement the new design

aub committed:
fix

aub committed:
add terms and conditions link to user profile

bshyong committed:
fix url name and remove duplicated code

bshyong committed:
Merge branch 'acct' of https://github.com/department-of-veterans-affairs/vets-website into acct

bshyong commented on src/js/common/components/RequiredTermsAcceptanceView.jsx:
```diff
@@ -0,0 +1,56 @@
+class RequiredTermsAcceptanceView extends React.Component {
+  componentWillMount() {
+    // I get errors and the app crashes if this line is called.
+    // this.props.checkAcceptance();
```
@aub what is the error that you see? This line kicks off the check which should ping the API for the acceptance check. It definitely should not crash so if you have more context I can see what's going on. 

My app doesn't crash but I get a 404 for the check which I presume means that the user has not accepted the terms.

```
code: "404"
detail: "The record identified by mhvac could not be found"
```

aub replied:
@bshyong I think the crash may be fixed. It still bails when getting the 404, and I think that's the appropriate response from the backend, so we'll need to handle that on the client. You're right that the 404 does indicate that the user hasn't accepted the terms.

bshyong committed:
uncomment code

aub committed:
show the terms dialog from preferences

bshyong committed:
terms content fetching

bshyong committed:
merge from upstream

bshyong committed:
null check for terms content

bshyong committed:
pull terms content from API

bshyong commented:
Great. I pushed some changes to pull the content from the API, so you'll need to seed your local database for it to work. The title needs to be 'mhvac' (that tripped me up a little bit when the API was unable to find the terms and conditions that I saved in the database)

The larger content fields take HTML and render it dangerously in the page to preserve formatting.
________________________________
From: Aubrey Holland <notifications@github.com>
Sent: Friday, May 19, 2017 8:45:43 AM
To: department-of-veterans-affairs/vets-website
Cc: Ben; Mention
Subject: Re: [department-of-veterans-affairs/vets-website] WIP: Start working on MHV account creation and terms and conditions accepting (#5571)


@aub commented on this pull request.

________________________________

In src/js/common/components/RequiredTermsAcceptanceView.jsx<https://github.com/department-of-veterans-affairs/vets-website/pull/5571#discussion_r117512537>:

> @@ -0,0 +1,56 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+
+import { checkAcceptance } from '../../user-profile/actions';
+
+import AcceptTermsPrompt from './AcceptTermsPrompt';
+import LoadingIndicator from '../../common/components/LoadingIndicator';
+
+class RequiredTermsAcceptanceView extends React.Component {
+  componentWillMount() {
+    // I get errors and the app crashes if this line is called.
+    // this.props.checkAcceptance();


@bshyong<https://github.com/bshyong> I think the crash may be fixed. It still bails when getting the 404, and I think that's the appropriate response from the backend, so we'll need to handle that on the client. You're right that the 404 does indicate that the user hasn't accepted the terms.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/department-of-veterans-affairs/vets-website/pull/5571#discussion_r117512537>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AASguQT6oVX-kmwzo-nKpWdjgxmgf7bPks5r7bkngaJpZM4Neagg>.

aub committed:
allow configurable terms names

bshyong committed:
Merge branch 'acct' of https://github.com/department-of-veterans-affairs/vets-website into acct

bshyong committed:
properly fetch and use terms acceptance objects

bshyong committed:
make cancel link go back to healthcare landing page

bshyong committed:
add mocks for account creation

bshyong commented:
We are passing specs! Also, currently MHV terms and conditions is only turned on in front of RX, but can be turned on for Secure Messaging as well

aub committed:
add terms to other MHV apps

aub commented:
@bshyong I added the check to SM and BB, so I think we're good there. Looks like we need to do some refactoring of JS in order to make code climate happy, but looks great otherwise. Would you mind taking a look at that?

bshyong commented:
Couple of things I will do here:
- Fix specs for Secure Messaging. The latest commit broke tests because the Terms components blocks access to the app
- Refactor some common code 

We will also need to plan for a demo/deployment strategy because this PR introduces a major change that will block users from the app.

bshyong committed:
light refactor of api client to minimize duplication

bshyong committed:
hook in acct creation helper to pass rx/sm specs

bshyong committed:
pass specs and improve logic for terms acceptance checking

bshyong committed:
Merge branch 'master' of https://github.com/department-of-veterans-affairs/vets-website into acct

bshyong committed:
Account Creation tests (#5606)

* tweak acct creation helper; add accept terms unit tests

* add unit tests for RequiredTermsAcceptanceView

bshyong committed:
fix bug for users who havent accepted terms

bshyong committed:
styling changes for terms acceptance component (#5692)

bshyong committed:
Add breadcrumbs to RX on terms and conditions page (#5688)

* Update home page layout to use navigation-links-lists.

Closes department-of-veterans-affairs/vets.gov-team#2555

* Update home page layout to use navigation-links-lists.

Closes department-of-veterans-affairs/vets.gov-team#2555

* Reduces hero image size. Updates tagline.

Closes department-of-veterans-affairs/vets.gov-team#2553

- Reordered partials so that VA variables override USWDS ones.

* Updating class names to use homepage-hero.

* Change failed submission language

* Update quick link styles

Closes department-of-veterans-affairs/vets.gov-team#2554

* Revert index.md for now. Need other PRs to be merged first.

* Re-adds new quick links style to index with reverted majorlinks

* Update home page layout to use navigation-links-lists.

Closes department-of-veterans-affairs/vets.gov-team#2555

* Add postinstall script for Heroku

This change should address the ENOENT errors we are seeing on Heroku

* Add divider to .va-h-ruled--stars (#5474)

* Add divider to .va-h-ruled--stars

Related to department-of-veterans-affairs/vets.gov-team#2555

* Removing header text.

* Adjust spacing of dividers, image size.

Fix mobile margin wonkiness on "Get started with..." header.

* Add divider to .va-h-ruled--stars

Related to department-of-veterans-affairs/vets.gov-team#2555

* Removing header text.

* Adjust spacing of dividers, image size.

Fix mobile margin wonkiness on "Get started with..." header.

* Fix 5490 logic for `currentSameAsPrevious` (#5493)

* Fix 5490 logic for `currentSameAsPrevious`

The 5490 helper only tried to copy of the `previousBenefit`'s
`veteranSocialSecurityNumber`, but wasn't including `view:veteranId`
in the path to get that. This change copies over the entire
`view:veteranId` to the `currentSponsorInformation` when
`currentSameAsPrevious` is set.

* Add a test for currentSameAsPrevious=false

* Fix PropType warnings (#5496)

* Fixing layout regression.

* Fixing regression on home page caused by merging other branches.

* Restoring line break on mobile.

* Make two-column link layout into a single column.

- Delete content/includes/navigation-links-list.html (duplicate)

* Refactoring the stars divider for better layout.

* Update background image.

* Update hero image, hero height.

* Adjusting gutters.

* Update home page layout to use navigation-links-lists.

Closes department-of-veterans-affairs/vets.gov-team#2555

* Update home page layout to use navigation-links-lists.

Closes department-of-veterans-affairs/vets.gov-team#2555

* Reduces hero image size. Updates tagline.

Closes department-of-veterans-affairs/vets.gov-team#2553

- Reordered partials so that VA variables override USWDS ones.

* Updating class names to use homepage-hero.

* Update quick link styles

Closes department-of-veterans-affairs/vets.gov-team#2554

* Revert index.md for now. Need other PRs to be merged first.

* Re-adds new quick links style to index with reverted majorlinks

* Update home page layout to use navigation-links-lists.

Closes department-of-veterans-affairs/vets.gov-team#2555

* Add divider to .va-h-ruled--stars (#5474)

* Add divider to .va-h-ruled--stars

Related to department-of-veterans-affairs/vets.gov-team#2555

* Removing header text.

* Adjust spacing of dividers, image size.

Fix mobile margin wonkiness on "Get started with..." header.

* Add divider to .va-h-ruled--stars

Related to department-of-veterans-affairs/vets.gov-team#2555

* Removing header text.

* Adjust spacing of dividers, image size.

Fix mobile margin wonkiness on "Get started with..." header.

* Fixing layout regression.

* Fixing regression on home page caused by merging other branches.

* Restoring line break on mobile.

* Make two-column link layout into a single column.

- Delete content/includes/navigation-links-list.html (duplicate)

* Refactoring the stars divider for better layout.

* Update background image.

* Update hero image, hero height.

* Adjusting gutters.

* Fix breakpoint-related layout bugs.

- Fix wrapping, placement of hero title at uncommon viewports.
- Fix layout of quicklinks

* Change 'Check your claim' to a Veterans Crisis Line trigger.

* Prevent body from scrolling when overlay is open.

* removing content from page (#5599)

so it won't be confused with the real page, /apply

* Update timeline styles on claim status (#5603)

* Change postal code to zip code for USA addresses on HCA (#5598)

* Deprecate old selectDropdown helper (#5580)

* merge 508 and e2e test for facility locator (#5596)

* merge 508 and e2e test for facility locator

* add new mocks and helpers for facility locator tests

* clean up mock data responses

* tweak tests

* fix incorrect object path

* Removes page from the SET_DATA action (#5605)

* Add empty pensions app (#5602)

* HCA fixes from testing (#5611)

* Spacing tweaks in new content templates (#5595)

* improved spacing within link lists

* Reduce link list heading font size

* WIP: Edu apply wizard (#5577)

* Relocate and conditionally load script for education/apply/

* Toggle child question visibility and value based on parent response

* Map question path using data-* attributes

* Implement toggle logic

* Update styling

* Incorporate feedback

* Remove unused styles

* Move wizard chagnes to copies

* Create wizard style module

* Update file structure for easy user testing

* Create alternate dataset for inputs to simplify hide logic

* Display warning for non-transfers

* Lint styles

* Refactor styles

* Refactor styles

* Create e2e test

* Resolve style name collisions

* Hide shame replacement

* Resolve merge conflict

* Update test

* Lint scss

* Update hca-rjsf.scss

Resolve strange linting issue (  3:2  error  Parsing error: Unexpected token) and correspond with master in editor.

* Remove swap shame from style

* Add a build type check

* Add /education/apply-wizard.md to the ignore list in build.js

* Use event delegation and refine warning toggle conditions

* Lint wizard logic

* Replace navigate away from logic

* Make test conditional on build type

* Remove wildcard from education/apply-wizard/ in ignoreList

* Revert qualifying elements for class selectors

* Remove linter-required returns and return values

* HCA rjsf timestamp fix (#5613)

* Actually fix the timestamp?

* Fix test

* Added 'View details' link and updated spacing between elements in GIBCT search results card. (#5623)

* Add vet and spouse financial disclosure pages (#5612)

* Add auth header to schemaform submissions (#5624)

* search when input is blurred (#5625)

* Search from landing page when blurred or suggestion is selected. (#5626)

* Rx Track Packages (#5576)

* Reformatted order history table to include list of prescriptions included in package.

* Added sorting logic to order history table.

* Split prescription page into details and track package tabs.

* Minor refactoring.

* Tweaked some visuals to match mocks.

* Updated track package links to direct to tab content.

* Left aligned call provider action in list version of Refill Prescriptions page.

* Renamed Prescription to PrescriptionCard for clarity.

* Removed unnecessary class name.

* Updated unit tests.

* Show column names next to fields in mobile version of track packages.

* Updated breadcrumbs behavior for Track Package tab.

* Fixed E2E tests.

* Updated mock tracking data and E2E test.

* Updated column name.

* Updated column name in mobile version.

* Show messages for unavailable tracking info. Unit tests for TrackPackage. (#5609)

* Show different messages in track package tab depending on availability of tracking info and refill status.

* Initial TrackPackage unit tests.

* Added unit tests for messages when tracking info is unavailable.

* Added check for URL.

* Removed commented line.

* Default to list view for 'Refill Prescriptions' page. (#5615)

* Default to list view for 'Refill Prescriptions' page.

* Corrected unintended change to limit to card view for small screens.

* Updated Rx E2E tests.

* simplify logic to resolve bugs (#5630)

* Scaffolding for Post-9/11 GI Bill status (#5610)

* Scaffolding for Post-9/11 GI Bill status

* Update path names to include the word "status"

* Update warning text and link (#5631)

* HCA rjsf fixes from testing in staging (#5632)

* Include chapter 1 - applicant information in pensions config (#5629)

* Include chapter 1 - applicant information in pensions config using definitions copied from education and relocated to common

* Create shared utils/labels and update original

* Replace references to original definitions

* Relocate unit tests

* Create pages folder and relocate applicantInformation there

* Create unit test

* Specify required fields

* Delete duplicate personId

* Update schema

* Update test

* Use local version of leaflet.css (#5634)

* upgrade leaflet

* use local copy of leaflet js

* upgrade react-leaflet

* upgrade react tabs; ensure FL is working

* address scss lint issues

* yarn install

* Fix bad variable override (#5635)

* removing caregiver info (#5636)

* Add financial disclosure pages for dependents (#5627)

* 5490 content change (#5641)

* Bold save information and capitalize benefits

* Change Benefit relinquishment to 'Benefits relinquishment' (plural on Benefits)

* Put warning text into a warning box

* Add message indicating that the comprehensive provision of information will expedite the approval process

* Make box blue and only bold first line

* Retain subdued styling of warning

* Add a break between info box and subsequent text

* removing email option (#5642)

* Additional error handling for FL (#5639)

* add error handling to fl

* Add facility locator error state; handle searches on back navigation

* address PR feedback

* add explanantory comment

* 1990e content change (#5645)

* Boldface work cannot be saved alert text

* Rephrase advisor recommendation box

* Change child label to include adopted and stepchild

* Switch to more restrictive ssn validation on forms (#5654)

* Relocate message from FormPage to applicantInformation uiSchema (#5653)

* Relocate message from FormPage to applicantInformation uiSchema

* Include strong tags and relabel as jsx file

* Add general military history page to pensions (#5644)

* Fix accessibility issues (#5650)

* remove extraneous search script

* address sass lint warning

* Reference correct attribute for 'Last submit date' in list view and allow sorting by it in card view. (#5651)

* Reference correct attribute for 'Last submit date' field in active prescriptions table.

* Fixed cut off prescription name title in cards.

* Renamed field for consistency.

* Moved sort menu to PrescriptionList component. Added last fill date option for consistency.

* Adds E2e tests for Kudos LOA3-protected apps (#5648)

* add login tests for rx

* add unauthed tests to messaging app

* add unauthed test for health-records

* add unauthed test for disability benefits app

* refactor loa1 and unauthed tests for clarity

* Fix info message on forms (#5657)

* Add disability history to pensions form (#5655)

* Add command line to run e2e tests against wcag2a accessibility rules (#5656)

* Add strict axeCheck

* Add command

* More failed acrobatics

* Add command line to test against wcag2a rules

* Read rules from config parameter before checking nightwatch globals

* Add additional sources fields to pensions form (#5638)

* Fixed broken SM headers in mobile. (#5664)

* Fixed broken SM headers in mobile.

* Updated settings header to match other headers.

* Suppress wcag2a rules for aXe checks due to aXe bugs (#5663)

* Default financial fields to 0 and make them optional (#5666)

* Improve form error handling and tracking (#5670)

* Add reducer tests for Facility Locator (#5661)

* upgrade mapbox; add facility reducer test

* add unit tests for search query reducer

* address PR feedback

* facility locator styling issues (#5672)

* add custom pin URLs to static maps (#5673)

* Disable email field when notifications are turned off (#5662)

* Handle API response from successful saving Rx preferences.

* Handle API response from successful saving SM preferences.

* Disable email input when notifications are turned off.

* Updated unit tests.

* Hide email input when turning off email notifications.

* Added tests for hiding input when notifications are off.

* Simplified props in Rx settings test.

* Removed unnecessary line breaks.

* adding adobe disclaimer (#5665)

* Pensions reserve and national guard (#5669)

* Provide national guard page

* WIP

* Include date

* Update vets-json-schema dependency version

* Add valid active service test case

* PR comments

* PR comments

* Remove unnecessary defaultDefinition-address

* Content updates for 1990 and 1995 (#5671)

* Conditional logic fix for incomplete updates (#5679)

* conditional logic fix for incomplete updates

* IE11 blob download support

* fix behavior of popup window for IE11 on bb (#5681)

* Default children array in hca rjsf (#5678)

* Go live changes for hca rjsf (#5660)

* Add employment history to pensions form (#5659)

* Update number of days to complete disability claim on 2 pages (#5683)

* Update apply.md

Update number of days per Jennifer Rudisil.

* number of days to complete disability claim

* fix URL for facility detail page map (#5684)

* Improve review page performance for hca (#5675)

* padding fix for terms

* fix padding

* pass unit tests

patrickvinograd committed:
Frontend fixes to support account creation soft launch

patrickvinograd committed:
Beta activation endpoint

bshyong committed:
merge master

patrickvinograd committed:
Review feedback

bshyong committed:
Fix terms model on user profile page (#5711)

bshyong committed:
Merge branch 'master' of https://github.com/department-of-veterans-affairs/vets-website into acct

bshyong committed:
Merge branch 'acct' of https://github.com/department-of-veterans-affairs/vets-website into acct

patrickvinograd committed:
Pv terms soft launch (#5707)

* Frontend fixes to support account creation soft launch

* Beta activation endpoint

bshyong committed:
Merge branch 'master' of https://github.com/department-of-veterans-affairs/vets-website into acct

bshyong committed:
Fix refresh error in ErrorView

bshyong committed:
Merge branch 'master' of https://github.com/department-of-veterans-affairs/vets-website into acct

patrickvinograd committed:
Update health beta registration API request path

patrickvinograd committed:
Reconcile profile terms with other terms updates

patrickvinograd committed:
Fix apparent mis-merge

patrickvinograd committed:
Merge pull request #5814 from department-of-veterans-affairs/pv-acct-updates

Pv account creation updates.

aub committed:
stop early terms loading and make cancel path configurable

patrickvinograd committed:
Merge pull request #5818 from department-of-veterans-affairs/tweakin

Terms of service tweaks

patrickvinograd committed:
Merge branch 'master' into acct

patrickvinograd committed:
Merge pull request #5826 from department-of-veterans-affairs/acct-master-20170621

Acct master 20170621

patrickvinograd committed:
Avoid presenting a no-op modal for terms and conditions (#5836)

patrickvinograd committed:
Merge branch 'master' into acct-master-20170628

patrickvinograd committed:
Consistent errorView guard in all MHV-based apps (#5903)

* Avoid parallel requests when loading messaging app

* Consistent ErrorViews for account creation errors across MHV apps

* Rework serial loading of folders/recipients

* Missed prop map

patrickvinograd committed:
Merge branch 'acct' into acct-master-20170628

patrickvinograd committed:
Merge pull request #5904 from department-of-veterans-affairs/acct-master-20170628

Acct master 20170628

bshyong reviewed (approved):
LGTM

patrickvinograd committed:
Merge branch 'master' into acct
